devtools-ci-index.yaml

```diff
@@ -382,13 +382,16 @@
     svc_name: none
     prj_name: dsaas-preview
     saas_git: none
+    saas_service_name: none
     saasherder_prepare: |
         if [ "$SAAS_GIT" != "none" ]; then
             SAAS_ENV=staging
             PATH=$PATH:~/.local/bin/
             CONTEXT=$(echo $SAAS_GIT | sed 's/.*://')
             SAAS_GIT=$(echo $SAAS_GIT | sed 's/:.*//')
-            SAAS_SERVICE_NAME=$GIT_REPO
+            if [ -z "$SAAS_SERVICE_NAME" ] || [ "$SAAS_SERVICE_NAME" == "none"]; then 
+                SAAS_SERVICE_NAME=$GIT_REPO
+            fi
```

Comment:
I believe this is a sysntax error. How about this:

```bash
if [[ -z "$SAAS_SERVICE_NAME" || "$SAAS_SERVICE_NAME" == "none" ]];  then
    SAAS_SERVICE_NAME=$GIT_REPO
fi
```

Reply:
could you explain why `if [ -z "$SAAS_SERVICE_NAME" ] || [ "$SAAS_SERVICE_NAME" == "none"]; then` is syntax error ? why there is a need of extra [] ?

Reply:
So, the problem is that in line 392 you have a typo, `"none"]` should be `"none" ]` (note the space between "none" and `]`).

The main reason I suggested `[[` is because from what I can see in the rest of the document it's what is typically used as opposed to single `[`.

However, to further expand why `[[` instead of `[`:

`[` is a binary:

```
$ which [
/usr/bin/[

$ file /usr/bin/[
/usr/bin/[: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked (uses shared libs), for 
GNU/Linux 2.6.32, BuildID[sha1]=564b25a9c94d9614f853bdc90fb7b2d7b8d4b800, stripped
```

So you are spawning two subprocess in your `if` statement. `[[` on the other hand is built into bash and it has a lot of features and easier syntax. Take a look at the `[[` from `man bash`.

Reply:
@jmelis ah, yeah good catch  about missing space